File: C/Reverse_Array.c
#include <stdio.h>

// Function to reverse array elements in place
void reverseArray(int arr[], int n) {
    int start = 0, end = n - 1, temp;
    while (start < end) {
        // Swap elements at start and end
        temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;
        start++;
        end--;
    }
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int n = sizeof(arr) / sizeof(arr[0]);

    reverseArray(arr, n);

    printf("Reversed Array: ");
    for (int i = 0; i < n; i++)
        printf("%d ", arr[i]);

    return 0;
}

File: rust/reverse_array.rs
fn main() {
    let mut arr = [1, 2, 3, 4, 5];
    let n = arr.len();

    for i in 0..n / 2 {
        arr.swap(i, n - i - 1);
    }

    println!("Reversed Array: {:?}", arr);
}
